packages/testcontainers/src/wait-strategies/http-wait-strategy.ts

```diff
@@ -66,14 +74,26 @@ export class HttpWaitStrategy extends AbstractWaitStrategy {
 
   public async waitUntilReady(container: Dockerode.Container, boundPorts: BoundPorts): Promise<void> {
     log.debug(`Waiting for HTTP...`, { containerId: container.id });
+
+    const exitStatus = "exited";
+    let containerExited = false;
     const client = await getContainerRuntimeClient();
+    const {abortOnContainerExit} = this.options || {};
 
     await new IntervalRetry<Response | undefined, Error>(this.readTimeout).retryUntil(
       async () => {
         try {
           const url = `${this.protocol}://${client.info.containerRuntime.host}:${boundPorts.getBinding(this.port)}${
             this.path
           }`;
+          const containerStatus = (await client.container.inspect(container)).State.Status;
```

Comment:
Perhaps we should put this in an `if (this.abortOnContainerExit)`, otherwise we are making many inspect calls for no reason

Reply:
Added if condition before inspect